File: test-runtimes/python/postgres.py
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.
#
import psycopg

def main(args):

    response = {"body": {}}

    with psycopg.connect(args.get("postgres_url")) as conn:

        # Open a cursor to perform database operations
        with conn.cursor() as cur:

            # Execute a command: this creates a new table
            cur.execute("""
                CREATE EXTENSION IF NOT EXISTS "pgcrypto";
                CREATE TABLE IF NOT EXISTS nuvolaris_table (
                    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
                    message varchar(100)        
                );
                """)

            # Pass data to fill a query placeholders and let Psycopg perform
            # the correct conversion (no SQL injections!)
            cur.execute("INSERT INTO nuvolaris_table(message) VALUES(%(message)s)",{"message":"Nuvolaris Postgres is up and running!"})
            # Make the changes to the database persistent

            # Query the database and obtain data as Python objects.
            cur.execute("SELECT message FROM nuvolaris_table")
            record = cur.fetchone()[0]
            response["body"] = record
            conn.commit()

    return response

File: stress-test/postgres.js
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
const { Client } = require('pg')

async function main(args) {
  console.log(args.postgres_url);
  const client = new Client({ connectionString: args.postgres_url });

  const createTableText = `
    CREATE EXTENSION IF NOT EXISTS "pgcrypto";
    CREATE TABLE IF NOT EXISTS nuvolaris_table (
        id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
        message varchar(100)        
    );
    `

  // Connect to database server
  await client.connect();

  response = { body: {} }

  try {
    await client.query(createTableText)
    const message = (Math.random() + 1).toString(36).substring(2);
    await client.query('INSERT INTO nuvolaris_table(message) VALUES($1)', [message])
    const { rows } = await client.query('SELECT id, message FROM nuvolaris_table where message = $1', [message]);
    const { rowCount } = await client.query('DELETE FROM nuvolaris_table WHERE id = $1', [rows[0].id]);
    response.body = {
      rows,
      rowCount
    }

  } catch (e) {
    console.log(e);
    throw e
  } finally {
    client.end();
  }

  return response;
}